chainer/optimizer_hooks/gradient_lars.py

```diff
@@ -0,0 +1,83 @@
+from chainer import cuda
+
+
+class GradientLARS(object):
+
+    """Optimizer/UpdateRule hook function for layer wise adaptive rate scaling.
+
+    See: `Large Batch Training of Convolutional Networks \
+          <https://arxiv.org/abs/1708.03888>`_
+       : `Convergence Analysis of Gradient Descent Algorithms \
+          with Proportional Updates \
+          <https://arxiv.org/abs/1801.03137>`_
+
+    This hook function scales all gradient arrays to fit to the weight norm.
+
+    In <https://arxiv.org/abs/1708.03888>
+        \\Delta w_{t+1} = m * \\Delta w_t
+                        + \\gamma * \\lambda * (\\nabla L_t(w) + \\beta w_t)
+        where
+           \\gamma : learning_rate
+           m       : momentum
+           \\beta  : weight_decay
+           \\eta   : lars_coeeficient
+           \\lamda : local_lr
+                     = \\eta * \\frac{\|w\|}{\|\\nabla L(w)\| + \\beta * \|w\|}
+    In GradientLARS
+        lr : lr in chainer.optimizers.SGD or chainer.optimizers.MomentumSGD
+        clip_rate : \\frac{\|w\|}{\|\\nabla L(w)\| + \\beta * \|w\|}
+    Then
+        \\gamma * \\lamda in arXiv = lr * clip_rate in GradientLARS and
```

Comment:
IMO, it would be less obvious where does this formula come from. So I would suggest, some thing like this:

As `lr` corresponds to `lambda * eta`, we define `clip_rate` as `\frac{|w|}{\|nabla L(w)| + \beta * |w|}` and reformulate the aforementioned formula as:
`\Delta w_{t+1} = m * \Delta w_t + lr * clip_rate * (\nabla L_t(w) + \beta w_t)`
and implement in this way.